project_status_report/models/project_status.py

```diff
@@ -0,0 +1,172 @@
+    name = fields.Char(compute='_compute_name',
+                       store=True,
+                       readonly=True,
+                       states={'draft': [('readonly', False)]})
+    cost_status = fields.Selection(selection=INDICATOR_STATUS,
+                                   readonly=True,
+                                   states={'draft': [('readonly', False)]})
+    cost_color = fields.Char(readonly=True,
+                             default='#00ff00',
+                             states={'draft': [('readonly', False)]})
+    cost_remarks = fields.Html(readonly=True,
+                               states={'draft': [('readonly', False)]})
+    quality_status = fields.Selection(selection=INDICATOR_STATUS,
+                                      readonly=True,
+                                      states={'draft': [('readonly', False)]})
+    quality_color = fields.Char(readonly=True,
+                                default='#00ff00',
+                                states={'draft': [('readonly', False)]})
+    quality_remarks = fields.Html(readonly=True,
+                                  states={'draft': [('readonly', False)]})
+    delay_status = fields.Selection(selection=INDICATOR_STATUS,
+                                    readonly=True,
+                                    states={'draft': [('readonly', False)]})
+    delay_color = fields.Char(readonly=True,
+                              default='#00ff00',
+                              states={'draft': [('readonly', False)]})
+    delay_remarks = fields.Html(readonly=True,
+                                states={'draft': [('readonly', False)]})
+    global_status = fields.Selection(selection=INDICATOR_STATUS,
+                                     readonly=True,
+                                     states={'draft': [('readonly', False)]})
+    global_color = fields.Char(readonly=True,
+                               default='#00ff00',
+                               states={'draft': [('readonly', False)]})
+    global_remarks = fields.Html(readonly=True,
+                                 states={'draft': [('readonly', False)]})
+    risks = fields.Html(readonly=True,
+                        states={'draft': [('readonly', False)]})
+    user_id = fields.Many2one(comodel_name='res.users',
+                              related='project_id.user_id',
+                              store=True,
+                              readonly=True)
+    indicator_ids = fields.One2many('project.status.indicator.value',
+                                    'report_id',
+                                    readonly=True,
+                                    states={'draft': [('readonly', False)]})
+    task_snapshot_ids = fields.One2many(
+        'project.task.snapshot',
+        'report_id',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+
+    @api.multi
+    def action_confirm(self):
+        self.state = 'ready'
```

Comment:
you should loop on `self` or use `ensure_one`

Reply:
or a write would do the trick and be usable on multiple records